src/iothread.c

```diff
@@ -452,6 +452,14 @@ int processClientsFromIOThread(IOThread *t) {
             continue;
         }
 
+        /* Run cron task for the client per second or it is marked as pending cron. */
+        if (c->last_cron_check_time + 1000 <= server.mstime ||
+            c->io_flags & CLIENT_IO_PENDING_CRON)
+        {
+            c->last_cron_check_time = server.mstime;
+            if (clientsCronRunClient(c)) continue;
+        }
```

Comment:
do we need to call updateClientMemUsageAndBucket() again after cron?
```suggestion
        /* Run cron task for the client per second or it is marked as pending cron. */
        if (c->last_cron_check_time + 1000 <= server.mstime ||
            c->io_flags & CLIENT_IO_PENDING_CRON)
        {
            c->last_cron_check_time = server.mstime;
            if (clientsCronRunClient(c)) continue;
        } else {
                /* Update the client in the mem usage */
                updateClientMemUsageAndBucket(c);
        }
```

Reply:
Not required at the technical level, just feel `clientsCronRunClient` calling `updateClientMemUsageAndBucket`  is not straight. i don't have strong opinion, if we do so, please update comment, such as
```
/* Update the client in the mem usage if clientsCronRunClient is not
 * being called, since that function already performs the update. */
 ```

Reply:
@wclmxxs would you like to do it?

Reply:
I'm concerned that the subsequent modifications might result in clientsCronRunClient returning 0 without updateClientMemUsageAndBucket being called.

Reply:
@wclmxxs unless the client was freed, otherwise updateClientMemUsageAndBucket() will be called.

Reply:
@sundb @ShooterIT I'm worried that someone will change the clientsCronRunClient function later, but to avoid a lot of repeated calls, I think it can be changed here, and I'll modify the code.

Reply:
LOL, that is what i worried about, it is not straight, so i asked to add comments above

Reply:
okay，i get it~

Reply:
do you have plan to do this

Reply:
Yes, I'll submit the changes in about three hours.

Reply:
@sundb @ShooterIT I have modified the code